message-index/messages/GHC-62016/deriving-int/index.md

```diff
@@ -0,0 +1,19 @@
+---
+title: Deriving Functor for a ground type
+---
+
+## Error message
+
+```
+Main.hs:5:12: error: [GHC-62016]
+    • Cannot derive well-kinded instance of form ‘Functor (Pet ...)’
+        Class ‘Functor’ expects an argument of kind ‘* -> *’
+    • In the data declaration for ‘Pet’
+```
+
+## Explanation
+
+The `Functor` type class applies to type constructors, that is,
+data types which take other type as parameter. The `Pet` data type
+defined in the code does *not* take any type parameters.
```

Comment:
```suggestion
The `Functor` type class applies to type constructors; that is,
data types which take one other type as a parameter. The `Pet` data type
defined in the code does *not* take any type parameters.
```

I'm not sure whether this should be "takes at least one other type"? or whether this is clear enough